src/main/java/org/veupathdb/service/eda/compute/plugins/correlation/CorrelationPlugin.java

```diff
@@ -0,0 +1,114 @@
+import org.veupathdb.service.eda.common.model.VariableDef;
+import org.veupathdb.service.eda.common.plugin.util.PluginUtil;
+import org.veupathdb.service.eda.compute.RServe;
+import org.veupathdb.service.eda.compute.plugins.AbstractPlugin;
+import org.veupathdb.service.eda.compute.plugins.PluginContext;
+import org.veupathdb.service.eda.generated.model.CorrelationComputeConfig;
+import org.veupathdb.service.eda.generated.model.CorrelationPluginRequest;
+import org.veupathdb.service.eda.generated.model.VariableSpec;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import static org.veupathdb.service.eda.common.plugin.util.PluginUtil.singleQuote;
+
+public class CorrelationPlugin extends AbstractPlugin<CorrelationPluginRequest, CorrelationComputeConfig> {
+
+  private static final String INPUT_DATA = "correlation_input";
+
+  public CorrelationPlugin(@NotNull PluginContext<CorrelationPluginRequest, CorrelationComputeConfig> context) {
+    super(context);
+  }
+
+  @NotNull
+  @Override
+  public List<StreamSpec> getStreamSpecs() {
+    return List.of(new StreamSpec(INPUT_DATA, getConfig().getCollectionVariable().getEntityId())
+        .addVars(getUtil().getChildrenVariables(getConfig().getCollectionVariable()))
+      );
+  }
+
+  @Override
+  protected void execute() {
+
+    CorrelationComputeConfig computeConfig = getConfig();
+    PluginUtil util = getUtil();
+    ReferenceMetadata meta = getContext().getReferenceMetadata();
+
+    String entityId = computeConfig.getCollectionVariable().getEntityId();
+    EntityDef entity = meta.getEntity(entityId).orElseThrow();
+    VariableDef computeEntityIdVarSpec = util.getEntityIdVarSpec(entityId);
+    String computeEntityIdColName = util.toColNameOrEmpty(computeEntityIdVarSpec);
+    String method = computeConfig.getCorrelationMethod().getValue();
+
+    // Get record id columns
+    List<VariableDef> idColumns = new ArrayList<>();
+    for (EntityDef ancestor : meta.getAncestors(entity)) {
+      idColumns.add(ancestor.getIdColumnDef());
+    }
+
+    HashMap<String, InputStream> dataStream = new HashMap<>();
+    dataStream.put(INPUT_DATA, getWorkspace().openStream(INPUT_DATA));
+    
+    RServe.useRConnectionWithRemoteFiles(dataStream, connection -> {
+      connection.voidEval("print('starting correlation computation')");
+
+      // Read in the abundance data
+      List<VariableSpec> computeInputVars = ListBuilder.asList(computeEntityIdVarSpec);
+      computeInputVars.addAll(util.getChildrenVariables(computeConfig.getCollectionVariable()));
+      computeInputVars.addAll(idColumns);
+      connection.voidEval(util.getVoidEvalFreadCommand(INPUT_DATA, computeInputVars));
+      connection.voidEval("abundanceData <- " + INPUT_DATA); // Renaming here so we can go get the sampleMetadata later
+
+      // Read in the sample metadata
+      // TO DO get only the continuous sample metadata!!
```

Comment:
feel like we should be introducing a util in edacommon for this

Reply:
like a 'grab all variables by predicate' helper..

Reply:
https://github.com/VEuPathDB/EdaCommon/issues/32